v2.0/partitioning.md

```diff
@@ -0,0 +1,543 @@
+~~~
+
+{% include copy-clipboard.html %}
+~~~ shell
+$ cockroach zone set roachlearn.students.graduated_us --insecure -f graduated_us.zone.yml
+~~~
+
+{% include copy-clipboard.html %}
+~~~ shell
+$ cockroach zone set roachlearn.students.current_au --insecure -f current_au.zone.yml
+~~~
+
+{% include copy-clipboard.html %}
+~~~ shell
+$ cockroach zone set roachlearn.students.graduated_au --insecure -f graduated_au.zone.yml
+~~~
+
+#### Step 6. Verify table partitions
+
+{% include copy-clipboard.html %}
+~~~ sql
+> show testing_ranges from table students;
+~~~
+
+You should see the following output:
+
+~~~ sql
++-----------------+-----------------+----------+----------+--------------+
+|    Start Key    |     End Key     | Range ID | Replicas | Lease Holder |
++-----------------+-----------------+----------+----------+--------------+
+| NULL            | /"AU"           |      260 | {1,2,3}  |            1 |
+| /"AU"           | /"AU"/17393     |      268 | {1,2,3}  |            1 |
+| /"AU"/17393     | /"AU"/PrefixEnd |      266 | {1,2,3}  |            1 |
+| /"AU"/PrefixEnd | /"CA"           |      267 | {1,2,3}  |            1 |
+| /"CA"           | /"CA"/17393     |      265 | {1,2,3}  |            1 |
+| /"CA"/17393     | /"CA"/PrefixEnd |      261 | {1,2,3}  |            1 |
+| /"CA"/PrefixEnd | /"NZ"           |      262 | {1,2,3}  |            3 |
+| /"NZ"           | /"NZ"/17393     |      284 | {1,2,3}  |            3 |
+| /"NZ"/17393     | /"NZ"/PrefixEnd |      282 | {1,2,3}  |            3 |
+| /"NZ"/PrefixEnd | /"US"           |      283 | {1,2,3}  |            3 |
+| /"US"           | /"US"/17393     |      281 | {1,2,3}  |            3 |
+| /"US"/17393     | /"US"/PrefixEnd |      263 | {1,2,3}  |            1 |
+| /"US"/PrefixEnd | NULL            |      264 | {1,2,3}  |            1 |
++-----------------+-----------------+----------+----------+--------------+
+(13 rows)
+
+Time: 11.586626ms
+~~~
+
+### Repartition a Table
+
+Consider the partitioned table of students of RoachLearn. Suppose the table has been partitioned on range to store the current students on fast and expensive storage devices (example: SSD) and store the data of the graduated students on slower, cheaper storage devices(example: HDD). Now suppose we want to change the date after which the students will be considered current to `2018-08-15`. We can achieve this by using the `ALTER TABLE` command.
+
+{% include copy-clipboard.html %}
+~~~ sql
+> ALTER TABLE students_by_range PARTITION BY RANGE (expected_graduation_date) (
+    PARTITION graduated VALUES FROM (MINVALUE) TO ('2018-08-15'),
+    PARTITION current VALUES FROM ('2018-08-15') TO (MAXVALUE));
+~~~
+
+### Unpartition a Table
+
+You can remove the partitions on a table by using the `PARTITION BY NOTHING` syntax with the `ALTER TABLE` command:
+
+{% include copy-clipboard.html %}
+~~~ sql
+> ALTER TABLE students PARTITION BY NOTHING;
+~~~
+
+## Locality–Resilience Tradeoff
```

Comment:
This section is hard to follow and feels a bit strange all the way at the end. But for now, perhaps we can just improve via bullets?

> - If only one replica is pinned to an Australian datacenter, then reads may be fast (via leases follow the workload) but writes will be slow. 
> - If two replicas are pinned to an Australian datacenter, then reads and writes will be fast (as long as the cross-ocean link has enough bandwidth that the third replica doesn’t fall behind). If those two replicas are in the same datacenter, then the loss of one datacenter can lead to data unavailability, so some deployments may want two separate Australian datacenters. 
> - If all three replicas are in Australian datacenters, then three Australian datacenters are needed to be resilient to a datacenter loss.